cmd/chartsvc/handler.go

```diff
@@ -61,34 +64,97 @@ type rel struct {
 	Links selfLink    `json:"links"`
 }
 
-// listCharts returns a list of charts
-func listCharts(w http.ResponseWriter, req *http.Request) {
+type meta struct {
+	TotalPages int `json:"totalPages"`
+}
+
+// getPageNumberAndSize extracts the page number and size of a request. Default (1, 0) if not set
+func getPageNumberAndSize(req *http.Request) (int, int) {
+	pageNumber := req.FormValue("page")
+	pageSize := req.FormValue("size")
+	numberInt, err := strconv.Atoi(pageNumber)
+	if err != nil || numberInt < 1 {
+		numberInt = 1
+	}
+	sizeInt, _ := strconv.Atoi(pageSize)
+	// Safeguard for negative numbers
+	if sizeInt < 0 {
+		sizeInt = 0
+	}
+	return numberInt, sizeInt
+}
+
+func min(a, b int) int {
+	if a < b {
+		return a
+	}
+	return b
+}
+
+func getChartList(repo string, pageNumber, pageSize int) (apiListResponse, interface{}, error) {
```

Comment:
`getPaginatedChartList`?

Reply:
Just a thought, I might be confused as to why `listCharts` doesn't implement this directly. `getPaginatedChartList` tells me that there is something more to this.

Reply:
fair enough, extra information won't hurt